huaweicloud/services/cpts/resource_huaweicloud_cpts_task.go

```diff
@@ -0,0 +1,293 @@
+		d.Set("name", response.Taskinfo.Name),
+		d.Set("project_id", response.Taskinfo.ProjectId),
+		d.Set("benchmark_concurrency", response.Taskinfo.BenchConcurrent),
+		d.Set("status", response.Taskinfo.RunStatus),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceTaskUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcCptsV3Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CPTS v1 client: %s", err)
+	}
+
+	id, err := strconv.ParseInt(d.Id(), 10, 32)
+	if err != nil {
+		return diag.Errorf("the task ID must be integer: %s", err)
+	}
+
+	projectId := int32(d.Get("project_id").(int))
+
+	if d.HasChanges("benchmark_concurrency", "name") {
+		benchConcurrent := int32(d.Get("benchmark_concurrency").(int))
+		_, err = client.UpdateTask(&model.UpdateTaskRequest{
+			TaskId: int32(id),
+			Body: &model.UpdateTaskRequestBody{
+				Id:              int32(id),
+				Name:            d.Get("name").(string),
+				ProjectId:       projectId,
+				BenchConcurrent: &benchConcurrent,
+			},
+		})
+
+		if err != nil {
+			return diag.Errorf("error updating the task %q: %s", id, err)
+		}
+	}
+
+	// Enable or stop task
+	if d.HasChange("operation") {
+		op := d.Get("operation").(string)
+		//Enable task
+		if op == operationEnable {
+			updateStatusRequest := model.UpdateTaskStatusRequest{
+				TestSuiteId: projectId,
+				TaskId:      int32(id),
+			}
+
+			clusterId := int32(d.Get("cluster_id").(int))
+			if clusterId > 0 {
+				updateStatusRequest.Body = &model.UpdateTaskStatusRequestBody{
+					ClusterType: clusterTypePrivate,
+					ClusterId:   clusterId,
+					Status:      runStatusToRunning,
+					NetworkInfo: &model.NetworkInfo{},
+				}
+			} else {
+				updateStatusRequest.Body = &model.UpdateTaskStatusRequestBody{
+					ClusterType: clusterTypeShare,
+					ClusterId:   0,
+					Status:      runStatusToRunning,
+					NetworkInfo: &model.NetworkInfo{
+						NetworkType: "internet",
+					},
+				}
+			}
+
+			_, err := client.UpdateTaskStatus(&updateStatusRequest)
+			if err != nil {
+				return diag.Errorf("error starting the task %q: %s", id, err)
+			}
+
+		}
+
+		// stop task
+		if op == operationStop {
+			_, err := client.UpdateTaskStatus(&model.UpdateTaskStatusRequest{
+				TestSuiteId: projectId,
+				TaskId:      int32(id),
+				Body: &model.UpdateTaskStatusRequestBody{
+					ClusterType: clusterTypeShare,
+					ClusterId:   -1,
+					Status:      runStatusFinished,
+					NetworkInfo: &model.NetworkInfo{},
+				},
+			})
+			if err != nil {
+				return diag.Errorf("error stopping the task %q: %s", id, err)
+			}
+		}
+
+		waitingforTaskFinished(ctx, client, id, d.Timeout(schema.TimeoutUpdate))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return resourceTaskRead(ctx, d, meta)
+}
+
+func resourceTaskDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
replace ctx with _ as it's not used in this func

Reply:
fix